src/sentry/api/serializers/models/incident.py

```diff
@@ -68,20 +69,18 @@ def get_attrs(self, item_list, user, **kwargs):
 
     def _get_incident_seen_list(self, incident, user):
         incident_seen = list(
-            IncidentSeen.objects.filter(incident=incident)
-            .select_related("user")
-            .order_by("-last_seen")
+            User.objects.filter(incidentseen__incident=incident).order_by(
+                "-incidentseen__last_seen"
+            )
         )
 
-        seen_by_list = []
         has_seen = False
 
         for seen_by in incident_seen:
-            if seen_by.user == user:
+            if seen_by.id == user.id:
                 has_seen = True
```

Comment:
Let's just rewrite this to `has_seen = any(seen_by for seen_by in incident_seen if seen_by.id = user.id))`